### PR TITLE
Fill the specified ref in webhook test payload

### PR DESCRIPTION
**Pull request by wolfogre**
The webhook payload should use the right ref when it‘s specified in the testing request.

The compare URL should not be empty, a URL like `compare/A...A` seems useless in most cases but is helpful when testing.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by wolfogre** [comment]
I think a `backport/v1.17` tag label is needed, and I will pick this pr to the `release/1.17` branch after it has been merged.

Could other maintainers please help to review it?

**Review by silverwind — APPROVED** [review]

File: routers/api/v1/repo/hook.go
Comment (wxiaoguang) [inline routers/api/v1/repo/hook.go]: IIRC, it's suggested to pass accurate ref name instead of guessing about the ref type @zeripath https://github.com/go-gitea/gitea/pull/20357#issuecomment-1184307624

Comment (wolfogre) [reply routers/api/v1/repo/hook.go]: 👍🏻I see, resolved it.

**Comment by zeripath** [comment]
Please send backport

**Pull request merged**
